classes/controlledVocab/ControlledVocabEntry.php

```diff
@@ -3,101 +3,199 @@
+     */
+    public function scopeWithControlledVocabId(Builder $query, int $controlledVocabId): Builder
+    {
+        return $query->where('controlled_vocab_id', $controlledVocabId);
     }
-}
 
-if (!PKP_STRICT_MODE) {
-    class_alias('\PKP\controlledVocab\ControlledVocabEntry', '\ControlledVocabEntry');
+    /**
+     * Scope a query to only include entries for a specific locales
+     */
+    public function scopeWithLocales(Builder $query, array $locales): Builder
+    {
+        return $query->whereExists(
+            fn ($query) => $query
+                ->select("{$this->primaryKey}")
+                ->from("{$this->getSettingsTable()}")
+                ->whereColumn(
+                    "{$this->getSettingsTable()}.{$this->primaryKey}",
+                    "{$this->getTable()}.{$this->primaryKey}"
+                )
+                ->whereIn(DB::raw("{$this->getSettingsTable()}.locale"), $locales)
+        );
+    }
+
+    /**
+     * Scope a query to only include entries for a specific setting name and values
+     */
+    public function scopeWithSettings(Builder $query, string $settingName, array $settingValue): Builder
+    {
+        return $query->whereExists(
+            fn ($query) => $query
+                ->select("{$this->primaryKey}")
+                ->from("{$this->getSettingsTable()}")
+                ->whereColumn(
+                    "{$this->getSettingsTable()}.{$this->primaryKey}",
+                    "{$this->getTable()}.{$this->primaryKey}"
+                )
+                ->where(
+                    "{$this->getSettingsTable()}.setting_name",
+                    '=',
+                    $settingName
+                )
+                ->whereIn(
+                    DB::raw("{$this->getSettingsTable()}.setting_value"),
+                    $settingValue
+                )
+        );
+    }
+
+    /**
+     * Scope a query to only include entries for a specific setting name and value with exact or partial match
+     */
+    public function scopeWithSetting(
+        Builder $query,
+        string $settingName,
+        string $settingValue,
+        ControlledVocabEntryMatch $match = ControlledVocabEntryMatch::PARTIAL
+    ): Builder
+    {
+        return $query->whereExists(
+            fn ($query) => $query
+                ->select("{$this->primaryKey}")
+                ->from("{$this->getSettingsTable()}")
+                ->whereColumn(
+                    "{$this->getSettingsTable()}.{$this->primaryKey}",
+                    "{$this->getTable()}.{$this->primaryKey}"
+                )
+                ->where(
+                    "{$this->getSettingsTable()}.setting_name",
+                    '=',
+                    $settingName
+                )
+                ->where(
+                    "{$this->getSettingsTable()}.setting_value",
+                    $match->operator(),
+                    $match->searchKeyword($settingValue)
+                )
+        );
+    }
```

Comment:
@asmecher as per your suggestion at https://github.com/pkp/pkp-lib/pull/10324#discussion_r1850814762, I have updated it handle string and array but that lead to 2 different implementation as we need to do a `IN` and `LIKE` based query . Wondering may be it's better to have one single `string|array` like implementation just for this case ?

Reply:
both are in `PKP\controlledVocab\ControlledVocabEntry`  also please see https://github.com/pkp/pkp-lib/pull/10324#discussion_r1888367467 .